### PR TITLE
add DeploymentConfigRollback object to openshift model

### PR DESCRIPTION
**Pull request by mluckam**
## Description

added DeploymentConfigRollback object to openshift model
https://docs.openshift.com/container-platform/4.10/rest_api/workloads_apis/deploymentconfigrollback-apps-openshift-io-v1.html

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [X] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [X] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [X] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@mluckam : Could you please add a test for this? See existing [tests](https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/ProjectHelmChartRepositoryTest.java).

**Comment by mluckam** [comment]
@rohanKanojia unit test has been added.

File: kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    //given
+    final DeploymentConfigRollback expected = mapper.readValue(
+      Helper.loadJson("/valid-deploymentConfigRollback.json"),
+      DeploymentConfigRollback.class);
+
+    //when
+    DeploymentConfigRollback actual = new DeploymentConfigRollbackBuilder()
+      .withName("app-to-rollback")
+      .withSpec(
+        new DeploymentConfigRollbackSpecBuilder().withFrom(
```
Comment (rohanKanojia) [inline kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java:45]: Is `.withNewSpec()` available here? 
```
  .withNewSpec()
     .withFrom( )
  .endSpec
```

File: kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            new ObjectReference(
```
Comment (rohanKanojia) [inline kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java:46]: Why not use ObjectReferenceBuilder here?

File: kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java (hunk 1 above)
Comment (mluckam) [reply kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java:45]: Yes it is.  This helps reduce the nesting. Thanks for the idea.

File: kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java (hunk 2 above)
Comment (mluckam) [reply kubernetes-model-generator/openshift-model/src/test/java/io/fabric8/openshift/api/model/DeploymentConfigRollbackTest.java:46]: Oversight, will update.

**Review by oscerd — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

**Comment by mluckam** [comment]
The DeploymentConfigRollback is utilized to rollback a DeploymentConfig via the API path "/apis/apps.openshift.io/v1/namespaces/{namespace}/deploymentconfigs/{name}/rollback", see [documentation](https://docs.openshift.com/container-platform/latest/rest_api/workloads_apis/deploymentconfigrollback-apps-openshift-io-v1.html#apisapps-openshift-iov1namespacesnamespacedeploymentconfigsnamerollback).  The model DeploymentConfigRollback was not currently in this project, but available in the [dependent model project](https://github.com/openshift/api).


Currently I am migrating to utilizing the kubernetes and openshift models from this project.  Once the models from this project are utilized, a next step would be to transition from interacting with the API via an HTTP client, to utilizing the OpenShiftClient.  This may be several months away though.  Until this migration occurs, I do not have plans to update the OpenShiftClient DSL.

**Comment by rohanKanojia** [comment]
@mluckam : Could you please create an issue for `/rollback` support in OpenShiftClient DSL?

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**